### PR TITLE
Migrate all repo files

### PR DESCRIPTION
**Pull request by CBroz1**
Superseding [this PR](https://github.com/datajoint/workflow-calcium-imaging-copy/pull/1)

- [X]  Remove templating CICD items that should only exist in element-array-ephys
- [X]  Add swp to gitignore for vim
- [X]  Notebooks
  - Remove non-applicable (00, 04, 05)
  - Revise 01, 02, 03, 06
  - Generate py scripts
  - Revise event csvs and ingestion script
- [X]  Remove non-applicable process.py
- [X]  Reduce docker compose dev/test to one version
- [X]  Full pytests

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
+contact_links:
+  - name: DataJoint Contribution Guideline
+    url: https://docs.datajoint.org/python/community/02-Contribute.html
```
Comment (kabilar) [inline .github/ISSUE_TEMPLATE/config.yml]:
```suggestion
    url: https://datajoint.com/docs/community/contribute/
```

File: docker/Dockerfile.test [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+USER anaconda:anaconda
+
+COPY ./workflow-optogenetics/docker/apt_requirements.txt /tmp/
+RUN /entrypoint.sh echo "Installed dependencies."
+
+WORKDIR /main/workflow-optogenetics
+
+# Always get interface/djarchive
+RUN pip install --no-deps "element-interface@git+https://github.com/datajoint/element-interface"
+RUN pip install --no-deps "djarchive-client@git+https://github.com/datajoint/djarchive-client"
```
Comment (kabilar) [inline docker/Dockerfile.test]:
```suggestion
```

Comment (kabilar) [reply docker/Dockerfile.test]: I don't think this is required for this workflow.

File: docker/Dockerfile.test [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+# ARG GITHUB_USERNAME=datajoint # tried moving to ENV
+USER anaconda:anaconda
+
+COPY ./workflow-optogenetics/docker/apt_requirements.txt /tmp/
+RUN /entrypoint.sh echo "Installed dependencies."
+
+WORKDIR /main/workflow-optogenetics
+
+# Always get interface/djarchive
```
Comment (kabilar) [inline docker/Dockerfile.test]:
```suggestion
```

File: docker/Dockerfile.test [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+RUN pip install --no-deps "element-interface@git+https://github.com/datajoint/element-interface"
```
Comment (kabilar) [inline docker/Dockerfile.test]:
```suggestion
```

Comment (kabilar) [reply docker/Dockerfile.test]: Can we add to `requirements.txt` and install from PyPI since this package is now published?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# DataJoint Workflow - Optogenetics
+
+The DataJoint Workflow for Optogenetics combines multiple DataJoint Elements for
+optogenetics research.  DataJoint Elements collectively standardize and automate data
+collection and analysis for neuroscience experiments.  Each Element is a modular
+pipeline for data storage and processing with corresponding database tables that can be
+combined with other Elements to assemble a fully functional pipeline.
+
+Installation and usage instructions can be found at the
+[Element documentation](datajoint.com/docs/elements/element-optogenetics).
```
Comment (kabilar) [inline README.md]:
```suggestion
Installation and usage instructions can be found at the
[Element documentation](https://datajoint.com/docs/elements/element-optogenetics).
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+![element-Optogenetics diagram](https://raw.githubusercontent.com/datajoint/element-Optogenetics/main/images/diagram_opto.svg)
```
Comment (kabilar) [inline README.md]:
```suggestion
![diagram](https://raw.githubusercontent.com/datajoint/element-optogenetics/main/images/diagram_flowchart.svg)
```

File: docker/setup.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+# If test cmd contains pytest, install
+if [[ "$TEST_CMD" == *pytest* ]]; then
+    pip install pytest
+    pip install pytest-cov
+fi
+
+# additional installs for running DLC
+pip install torch
+pip install ffmpeg
```
Comment (kabilar) [inline docker/setup.sh]:
```suggestion
```

File: docker/setup.sh [hunk 8]
```diff
@@ -0,0 +1,37 @@
+#! /bin/bash
+export $(grep -v '^#' /main/.env | xargs)
+
+echo "INSALL OPTION:" $INSTALL_OPTION
```
Comment (kabilar) [inline docker/setup.sh]:
```suggestion
echo "INSTALL OPTION:" $INSTALL_OPTION
```

File: docker/Dockerfile.test (hunk 4 above)
Comment (CBroz1) [reply docker/Dockerfile.test]: Oops - holdover from previous methods. The `requirements.txt` file does get interface from PyPI

File: tests/test_pipeline_generation.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    assert subject.Subject.full_table_name in surgery.Implantation.parents()
+
+
+def test_opto_pipeline(pipeline):
+    session = pipeline["session"]
+    surgery = pipeline["surgery"]
+    opto = pipeline["opto"]
+    Device = pipeline["Device"]
+
+    # test connection opto.VideoRec -> schema children
```
Comment (kabilar) [inline tests/test_pipeline_generation.py]:
```suggestion
    # test connection opto.OptoProtocol -> schema children
```

File: docker/docker-compose.yaml [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+  workflow:
+    networks:
+      optogenetics:
+    build:
+      context: ../../
+      dockerfile: ./workflow-optogenetics/docker/Dockerfile.test
+      args:
+        - GITHUB_USERNAME=${GITHUB_USERNAME}
+    image: workflow-optogenetics:0.0.0
```
Comment (kabilar) [inline docker/docker-compose.yaml]:
```suggestion
    image: workflow-optogenetics:0.1.0
```

File: tests/test_ingest.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
+"""Tests ingestion into schema tables: Lab, Subject, Session
+    1. Assert length of populating data conftest
+    2. Assert exact matches of inserted data fore key tables
```
Comment (kabilar) [inline tests/test_ingest.py]:
```suggestion
    2. Assert exact matches of inserted data for key tables
```

File: tests/test_ingest.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
+"""Tests ingestion into schema tables: Lab, Subject, Session
+    1. Assert length of populating data conftest
```
Comment (kabilar) [inline tests/test_ingest.py]:
```suggestion
    1. Assert length of data populated with conftest
```

File: docker/setup.sh [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+echo "INSTALL OPTION:" $INSTALL_OPTION
+cd /main/
+# all local installs, mapped from host
+if [ "$INSTALL_OPTION" == "local-all" ]; then
+    for f in lab animal session event optogenetics; do
+        pip install -e ./element-${f}
+    done
+    pip install -e ./workflow-optogenetics
+# all except workflow pip installed
```
Comment (kabilar) [inline docker/setup.sh]:
```suggestion
# Install all from GitHub except for (optionally) optogenetics Element and Workflow
```

File: tests/test_pipeline_generation.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    assert subject.Subject.full_table_name in surgery.Implantation.parents()
+
+
+def test_opto_pipeline(pipeline):
+    session = pipeline["session"]
+    surgery = pipeline["surgery"]
+    opto = pipeline["opto"]
+    Device = pipeline["Device"]
+
+    # test connection opto.OptoProtocol -> schema children
```
Comment (kabilar) [inline tests/test_pipeline_generation.py]:
```suggestion
    # test connection opto.OptoProtocol -> parents
```

File: notebooks/py_scripts/02-workflow-structure-optional.py [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+#
+
+import datajoint as dj
+
+from workflow_optogenetics.pipeline import Device, lab, opto, session, subject, surgery
+
+# Each module contains a schema object that enables interaction with the schema in the database.
+#
+
+# + Each module imported above corresponds to one schema inside the database. For example, `ephys` corresponds to `neuro_ephys` schema in the database.
```
Comment (kabilar) [inline notebooks/py_scripts/02-workflow-structure-optional.py]:
```suggestion
# + Each module imported above corresponds to one schema inside the database. For example, `opto` corresponds to `neuro_opto` schema in the database.
```

File: notebooks/py_scripts/02-workflow-structure-optional.py [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+opto.schema
+
+# -
+
+# The table classes in the module corresponds to a table in the schema in the database.
+#
+
+# + Each datajoint table class inside the module corresponds to a table inside the schema. For example, the class `ephys.EphysRecording` correponds to the table `_ephys_recording` in the schema `neuro_ephys` in the database.
```
Comment (kabilar) [inline notebooks/py_scripts/02-workflow-structure-optional.py]:
```suggestion
# + Each datajoint table class inside the module corresponds to a table inside the schema. For example, the class `opto.OptoWaveform` correponds to the table `_opto_waveform` in the schema `neuro_opto` in the database.
```

File: notebooks/py_scripts/02-workflow-structure-optional.py [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+# preview columns and contents in a table
+opto.OptoWaveform()
+
+
+# + The first time importing the modules, empty schemas and tables will be created in the database. [markdown]
+# By importing the modules for the first time, the schemas and tables will be created inside the database.
```
Comment (kabilar) [inline notebooks/py_scripts/02-workflow-structure-optional.py:56]:
```suggestion
```

Comment (kabilar) [reply notebooks/py_scripts/02-workflow-structure-optional.py:56]: Duplicate with the sentence above.

**Comment by kabilar** [comment]
Thanks @CBroz1!  Great work.  I will take over this work by closing this pull request and pulling from your fork.  I will add support for GitHub Codespaces, and remove the `ingest.py` script to follow the recent effort we are making to simplify all Workflows.

**Pull request closed without merging**